pkg/controllers/consolidation/types.go

```diff
@@ -20,70 +20,102 @@ import (
 
 	v1 "k8s.io/api/core/v1"
 
+	"github.com/aws/karpenter-core/pkg/apis/provisioning/v1alpha5"
 	"github.com/aws/karpenter/pkg/controllers/provisioning/scheduling"
 )
 
-// ProcessResult is used to indicate the result of consolidating so we can optimize by not trying to consolidate if
+// DeprovisioningResult is used to indicate the action of consolidating so we can optimize by not trying to consolidate if
 // we were unable to consolidate the cluster and it hasn't changed state with respect to pods/nodes.
-type ProcessResult byte
+type DeprovisioningResult byte
 
 const (
-	ProcessResultNothingToDo ProcessResult = iota
-	ProcessResultFailed
-	ProcessResultConsolidated
+	DeprovisioningResultNothingToDo DeprovisioningResult = iota // there are no actions that can be performed given the current cluster state
+	DeprovisioningResultRetry                                   // we attempted an action, but its validation failed so retry soon
+	DeprovisioningResultFailed                                  // the action failed entirely
+	DeprovisioningResultSuccess                                 // the action was successful
 )
 
-type consolidateResult byte
+func (r DeprovisioningResult) String() string {
+	switch r {
+	case DeprovisioningResultNothingToDo:
+		return "Nothing to do"
+	case DeprovisioningResultRetry:
+		return "Retry"
+	case DeprovisioningResultFailed:
+		return "Failed"
+	case DeprovisioningResultSuccess:
+		return "Success"
+	default:
+		return fmt.Sprintf("Unknown (%d)", r)
+	}
+}
+
+type lifecyleAction byte
 
 const (
-	consolidateResultUnknown consolidateResult = iota
-	consolidateResultNotPossible
-	consolidateResultDelete
-	consolidateResultDeleteEmpty
-	consolidateResultReplace
-	consolidateResultNoAction
+	lifeCycleActionUnknown lifecyleAction = iota
```

Comment:
Since you changed some of the types to be `DeprovisioningResult....` it'd make sense to change the actions as well.